libmscore/transpose.cpp

```diff
@@ -665,6 +665,18 @@ void Note::transposeDiatonic(int interval, bool keepAlterations, bool useDoubleA
       score()->undoChangePitch(this, newPitch, newTpc1, newTpc2);
       }
 
+//---------------------------------------------------------
+//   transposeDiatonicAlterations
+//---------------------------------------------------------
+
+void Score::transposeDiatonicAlterations(TransposeDirection direction)
+      {
+      // Transpose current selection diatonically (up/down) while keeping degree alterations
+      // Note: Score::transpose() absolutely requires valid selection before invocation.
+      if (!score()->selection().isNone())
+            score()->transpose(TransposeMode::DIATONICALLY, direction, Key::C, 1, true, true, true);
```

Comment:
No need to call `score()` in these two lines.

Reply:
Because of the `[this]`  in the cmd.cpp I guess?

Reply:
Oh yeah, that was from moving over to scoreview.cpp and seemingly requiring it over there and then back again but not foregoing it upon return.

Reply:
Well, the `[this]` in the cmd.cpp is needed to capture the Score object in order to call `Score::transposeDiatonicAlterations()`, but my point is that it is not necessary to call `score()` from within the Score class.

Reply:
@Jojo: I think it's because before I implemented a function to incorporate the [if clause], the scope was merely in scoreview.cpp's cmd() function which was of ScoreView (derived from MuseScoreView and QWidget), and in turn didn't have direct access to Score::selection() as a  result and needed rather indirect means to gain access to the score object through MuseScoreView's::score(). 

I'll remove the indirect calls in a moment.